SSA/Projects/InstCombine/LLVM/EDSL.lean

```diff
@@ -218,7 +218,7 @@ elab "[alive_icom (" mvars:term,* ")| " reg:mlir_region "]" : term => do
   synthesizeSyntheticMVarsNoPostponing
   let com : Q(MLIR.AST.ExceptM (MOp $φ) (Σ (Γ' : Ctxt (MTy $φ)) (ty : (MTy $φ)), Com (MOp $φ) Γ' ty)) ←
     withTheReader Core.Context (fun ctx => { ctx with options := ctx.options.setBool `smartUnfolding false }) do
-      withTransparency (mode := TransparencyMode.all) <|
+      withTransparency (mode := TransparencyMode.default) <|
```

Comment:
If so, I would personally suggest doing this as a separate commit.

Reply:
It's technically unrelated, but I did the change here because I want to prevent divergence between the generic meta-code, and the alive meta-code.
It indeed tunes down the unfolding, making it `all` in the first place was more aggressive than needed. We reduce to get rid of the `Transform` machinery, and none of that is `irreducible` (unless a dialect implements the typeclass with irreducible `def`s, we just shouldn't do that).

Reply:
I actually want to tone down the reduction even more, to just recursive `whnf`, similar to what I did for `change_mlir_context`